Source/Core/VideoBackends/D3D/TextureCache.cpp

```diff
@@ -179,17 +181,165 @@ void TextureCache::TCacheEntry::FromRenderTarget(u32 dstAddr, unsigned int dstFo
+	b = Convert5To8((val)& 0x1f);
+	a = 0xFF;
+	return float4(r, g, b, a) / 255;
+}
+
+float4 DecodePixel_IA8(uint val)
+{
+	int i = val & 0xFF;
+	int a = val >> 8;
+	return float4(i, i, i, a) / 255;
+}
+
+void main(
+	out float4 ocol0 : SV_Target,
+	in float4 pos : SV_Position,
+	in float3 uv0 : TEXCOORD0)
+{
+	uint src = round(Tex0.Sample(samp0,uv0) * Multiply).r;
+	src = Tex1.Load(src);
+	src = ((src << 8) & 0xFF00) | (src >> 8);
+	ocol0 = DECODE(src);
+}
+)HLSL";
+
+void TextureCache::ConvertTexture(TCacheEntryBase* entry, TCacheEntryBase* unconverted, void* palette, TlutFormat format)
+{
+	g_renderer->ResetAPIState();
+
+	// stretch picture with increased internal resolution
+	const D3D11_VIEWPORT vp = CD3D11_VIEWPORT(0.f, 0.f, (float)unconverted->config.width, (float)unconverted->config.height);
+	D3D::context->RSSetViewports(1, &vp);
+
+	D3D11_BOX box{ 0, 0, 0, 512, 1, 1 };
+	D3D::context->UpdateSubresource(palette_buf, 0, &box, palette, 0, 0);
+
+	D3D::stateman->SetTexture(1, palette_buf_srv);
+
+	float params[4] = { unconverted->format == 0 ? 15.f : 255.f };
+	D3D::context->UpdateSubresource(palette_uniform, 0, nullptr, &params, 0, 0);
+	D3D::stateman->SetPixelConstants(palette_uniform);
+
+	const D3D11_RECT sourcerect = CD3D11_RECT(0, 0, unconverted->config.width, unconverted->config.height);
+
+	D3D::SetPointCopySampler();
+
+	// Make sure we don't draw with the texture set as both a source and target.
+	// (This can happen because we don't unbind textures when we free them.)
+	D3D::stateman->UnsetTexture(static_cast<TCacheEntry*>(entry)->texture->GetSRV());
+
+	D3D::context->OMSetRenderTargets(1, &static_cast<TCacheEntry*>(entry)->texture->GetRTV(), nullptr);
+
+	// Create texture copy
+	D3D::drawShadedTexQuad(
+		static_cast<TCacheEntry*>(unconverted)->texture->GetSRV(),
+		&sourcerect, unconverted->config.width, unconverted->config.height,
+		palette_pixel_shader[format],
+		VertexShaderCache::GetSimpleVertexShader(), VertexShaderCache::GetSimpleInputLayout(),
+		GeometryShaderCache::GetCopyGeometryShader());
+
+	D3D::context->OMSetRenderTargets(1, &FramebufferManager::GetEFBColorTexture()->GetRTV(), FramebufferManager::GetEFBDepthTexture()->GetDSV());
+
+	g_renderer->RestoreAPIState();
+}
+
+ID3D11PixelShader *GetConvertShader(const char* Type)
+{
+	std::string shader = "#define DECODE DecodePixel_";
+	shader.append(Type);
+	shader.append("\n");
+	shader.append(palette_shader);
+	return D3D::CompileAndCreatePixelShader(shader);
+}
+
 TextureCache::TextureCache()
 {
 	// FIXME: Is it safe here?
 	g_encoder = new PSTextureEncoder;
 	g_encoder->Init();
+
+	palette_buf = nullptr;
+	palette_buf_srv = nullptr;
+	palette_uniform = nullptr;
+	palette_pixel_shader[GX_TL_IA8] = GetConvertShader("IA8");
+	palette_pixel_shader[GX_TL_RGB565] = GetConvertShader("RGB565");
+	palette_pixel_shader[GX_TL_RGB5A3] = GetConvertShader("RGB5A3");
+	auto lutBd = CD3D11_BUFFER_DESC(sizeof(u16) * 256, D3D11_BIND_SHADER_RESOURCE);
```

Comment:
It won't work.  I haven't seen anything try to use it.   Making it work would probably require changing the way we generate EFB copies.